### ISSUE TITLE
Cups returns invalid (empty) `media-supported` values

### ISSUE THREAD

**Issue by porridge** [OpenPrinting/cups#1078]
**Describe the bug**

I share my USB-connected printer using cups. Cups response to `get-printer-attributes` requests is invalid: a couple of `media-supported` values are empty. The problem is that recently downstream servers that exercise extra checks to guard against CVE-2024-47175 started to choke on this invalid response and refuse to choose the printer for printing.

**To Reproduce**
1. Connect the "HP Smart Tank 720" printer using USB cable
2. Set it up using the stock ppd file ([kotlownia2.ppd.txt](https://github.com/user-attachments/files/17343780/kotlownia2.ppd.txt))
3. Inspect it using `ipptool -j http://192.168.192.1:631/printers/kotlownia2 get-printer-attributes.test > via-cups.json` ([via-cups.json](https://github.com/user-attachments/files/17343678/via-cups.json))
4. Notice a couple of empty values in `media-supported` list in the `printer-attributes-tag` group, and the following warning from `ipptool`: `"media-supported": Bad keyword value "" - invalid character (RFC 8011 section 5.1.4).`

**Expected behavior**
A valid response.

**System Information:**
 - OS and its version: Debian GNU/Linux 12.7 (bookworm)
 - Application: another cups server, evince trying to use it
 - CUPS version 2.4.2-3+deb12u8

**Additional context**

More info in https://bugs.debian.org/cgi-bin/bugreport.cgi?bug=1084021

The above Debian bug report also shows that when the printer is contacted directly over its WiFi interface, it produces a valid get-printer-attributes response. [direct.json](https://github.com/user-attachments/files/17343689/direct.json)

Some more diagnostics:

```
$ sudo ipp-usb check
Configuration files: OK
IPP over USB devices:
 Num  Device              Vndr:Prod  Model
   1. Bus 002 Device 007  03f0:3a54  "HP Smart Tank 710-720 series"
```

```
$ lpstat -l -e
HP_Smart_Tank_710_720_series_FDDE68 network none ipps://HP%20Smart%20Tank%20710-720%20series%20%5BFDDE68%5D._ipps._tcp.local/
HP_Smart_Tank_710_720_series_FDDE68_USB network none ipp://HP%20Smart%20Tank%20710-720%20series%20%5BFDDE68%5D%20(USB)._ipp._tcp.local/
HP_Smart_Tank_710_720_series_piec network none ipps://HP%20Smart%20Tank%20710-720%20series%20%40%20piec._ipps._tcp.local/cups
kotlownia2 permanent ipp://localhost/printers/kotlownia2 usb://HP/Smart%20Tank%20710-720%20series?serial=CN172171F5&interface=1
```

```
$ lpoptions -p kotlownia2 -l
PageSize/Media Size: 2L L 4x6in 5x7in 8.5x13 8x10 A4 *A4.Duplex A5 A6 B5 B5.Duplex JB5 JB5.Duplex Custom1 Envelope#10 EnvelopeC6 EnvelopeDL Executive Executive.Duplex IndexCard4x6 IndexCard5x8 IndexCardA4 IndexCardLetter JapaneseEnvelope#3 JapaneseEnvelope#4 JapanesePostcard Legal Letter Letter.Duplex OfukuHagaki Statement Executive.FB Legal.FB Statement.FB B5.FB Envelope#10.FB EnvelopeDL.FB EnvelopeC6.FB JapaneseEnvelope#4.FB 8.5x13.FB Custom.WIDTHxHEIGHT
Duplex/Double-Sided Printing: *DuplexNoTumble DuplexTumble None
InputSlot/Media Source: *Main
ColorModel/Output Mode: CMYGray KGray *RGB
MediaType/Media Type: *Plain Glossy CoatedBrochure CoatedMatte BrochureGlosy FastGlossy Coated Brochure PlainPaperLight
OutputMode/Print Quality: *Normal Best Photo
```

[attr.log](https://github.com/user-attachments/files/17343755/attr.log) from this command:
```
    $ ipptool --ippserver attr.log -v ipp://localhost:60000/ipp/print get-printer-attributes.test
```

**Comment by porridge** [comment]
Thank you @michaelrsweet , the `#` characters indeed seemed to be what was tripping cups up. After performing the recommended replacements (and another for `JapaneseEnvelope#3`) things seem to work.

**However**, in the name of robustness, wouldn't it be better if going forward cups simply omitted problematic media in the response to `get-printer-attributes` rather than using invalid empty values? Arguably missing items would be a better hint that something is wrong than the current "printer unusable" situation.

**Comment by michaelrsweet** [comment]
@porridge

> However, in the name of robustness, wouldn't it be better if going forward cups simply omitted problematic media in the response to get-printer-attributes rather than using invalid empty values? Arguably missing items would be a better hint that something is wrong than the current "printer unusable" situation.

I'm pretty sure (still need to spend some time testing this) that the current CUPS 2.4.x code does, in fact, correctly convert such names into names/sizes that are valid in IPP. There have been a *lot* of changes and fixes in this code since CUPS 2.4.2 in May 2022...

**Comment by michaelrsweet** [comment]
... and it looks like the change was part of CUPS 2.4.11.
